### PR TITLE
Added Save + Load of Game State

### PR DESCRIPTION
**Pull request by po-trottier**
Merge #91 before looking at this PR as it's based on top of the other PR

Add Saving of the following:
- [x] Player (HP + transform)
- [x] Enemies (HP + transform + state)
- [x] Relics (Player has or not)
- [x] Keys (location + number of keys picked up)
- [x] Scene (current scene)

Add Loading of the following:
- [x] Player
- [x] Enemies
- [x] Relics
- [x] Keys
- [ ] Scene

Scene will be done when we implement the Scene Loading.

Known Bugs:
- The single enemy that's in the scene gets teleported through the wall if you save & load the scene without moving the enemy (if they remain in their spawn point, they'll be placed at the proper place on load but then when you unpause they teleport... Could be due to the rigidbody movement ?)

### CODE REVIEW

File: Hush/Assets/Plugins/PyroParticles/Prefab/Script/FireCollisionForwardScript.cs [hunk 1]
```diff
@@ -1,6 +1,4 @@
 ﻿using UnityEngine;
-using System.Collections;
-using Common;
 
 namespace DigitalRuby.PyroParticles
 {
```
Comment (daniela-venuta) [inline Hush/Assets/Plugins/PyroParticles/Prefab/Script/FireCollisionForwardScript.cs:4]: Why was this file changed?

**Review by daniela-venuta — APPROVED** [review]

File: Hush/Assets/Plugins/PyroParticles/Prefab/Script/FireCollisionForwardScript.cs (hunk 1 above)
Comment (po-trottier) [reply Hush/Assets/Plugins/PyroParticles/Prefab/Script/FireCollisionForwardScript.cs:4]: Remove useless imports + move custom logic to inherited class

**Review by emilie-martin — APPROVED** [review]
